govcd/vdc.go

```diff
@@ -1350,3 +1353,62 @@ func (vdc *Vdc) CloneVapp(sourceVapp *types.CloneVAppParams) (*VApp, error) {
 	err = vapp.Refresh()
 	return vapp, err
 }
+
+// Get the details of a hardware version
+func (vdc *Vdc) GetHardwareVersion(name string) (*types.VirtualHardwareVersion, error) {
+	found := false
+	for _, hwVersion := range vdc.Vdc.Capabilities[0].SupportedHardwareVersions.SupportedHardwareVersion {
+		if hwVersion.Name == name {
+			found = true
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("hardware version %s not found or not supported", name)
+	}
+
+	vdcHref, err := url.ParseRequestURI(vdc.Vdc.HREF)
+	if err != nil {
+		return nil, fmt.Errorf("error getting VDC href: %s", err)
+	}
+	vdcHref.Path += "/hwv/" + name
+
+	hardwareVersion := &types.VirtualHardwareVersion{}
+
+	_, err = vdc.client.ExecuteRequest(vdcHref.String(), http.MethodGet, types.MimeVirtualHardwareVersion, "error getting hardware version: %s", nil, hardwareVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	return hardwareVersion, nil
+}
+
+// Get highest supported hardware version of a VDC
+func (vdc *Vdc) GetHighestHardwareVersion() (*types.VirtualHardwareVersion, error) {
+	err := vdc.Refresh()
+	if err != nil {
+		return nil, err
+	}
+
+	hardwareVersions := vdc.Vdc.Capabilities[0].SupportedHardwareVersions.SupportedHardwareVersion
+	// Get last item (highest version) of SupportedHardwareVersions
+	highestVersion := hardwareVersions[len(hardwareVersions)-1].Name
+
+	hardwareVersion, err := vdc.GetHardwareVersion(highestVersion)
+	if err != nil {
+		return nil, err
+	}
+	return hardwareVersion, nil
+}
+
+// FindOsFromId attempts to find a OS by ID by the given hardware version
+func FindOsFromId(hardwareVersion *types.VirtualHardwareVersion, osId string) (*types.OperatingSystemInfoType, error) {
```

Comment:
For consideration only - not insisting but another thought - could this function be a method of some type or a few types? This looks to be orphaned now and is hard to find unless you know about it.

Reply:
Agree, assigned it to `*Vdc` as hardware versions are retrieved from VDCs